lightning/src/routing/scoring.rs

```diff
@@ -997,10 +997,10 @@ impl<L: Deref<Target = u64>, BRT: Deref<Target = HistoricalBucketRangeTracker>,
 	}
 
 	fn decayed_offset_msat(&self, offset_msat: u64) -> u64 {
-		self.now.duration_since(*self.last_updated).as_secs()
-			.checked_div(self.params.liquidity_offset_half_life.as_secs())
-			.and_then(|decays| offset_msat.checked_shr(decays as u32))
-			.unwrap_or(0)
+		let elapsed_time = self.now.duration_since(*self.last_updated).as_secs() as f64;
+		let half_life = self.params.liquidity_offset_half_life.as_secs() as f64;
+		let decays = elapsed_time / half_life;
+		(0.5f64.powf(decays) * offset_msat as f64) as u64
```

Comment:
fwiw, I guess you could also do 2f64.powf() and then use that as the divisor with some special checked_div function if there's any benefit to that, but I don't know whether there is one

Reply:
Looks a little slower that way (49-50ms).

Reply:
oh wow, that's actually a bunch. Thanks for checking

Reply:
That was the absolute runtime, FWIW. So 1-2ms slower than baseline (i.e., shifting).